src/main/java/io/cdap/plugin/gcp/bigquery/sink/DelegatingMultiSinkOutputCommitter.java

```diff
@@ -0,0 +1,176 @@
+/*
+ * Copyright © 2021 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.plugin.gcp.bigquery.sink;
+
+import io.cdap.cdap.api.data.schema.Schema;
+import org.apache.hadoop.mapreduce.JobContext;
+import org.apache.hadoop.mapreduce.JobStatus;
+import org.apache.hadoop.mapreduce.OutputCommitter;
+import org.apache.hadoop.mapreduce.TaskAttemptContext;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Output Committer which creates and delegates operations to other Bigquery Output Committer instances.
+ *
+ * Delegated instances are supplied along with a schema, which is used to configure the commit operation.
+ */
+public class DelegatingMultiSinkOutputCommitter extends OutputCommitter {
+  private final Map<String, OutputCommitter> committerMap;
+  private final Map<String, Schema> schemaMap;
+  private String projectName;
+  private String datasetName;
+  private String bucketName;
+  private String bucketPathUniqueId;
+
+  public DelegatingMultiSinkOutputCommitter() {
+    committerMap = new HashMap<>();
+    schemaMap = new HashMap<>();
+  }
+
+  public void setProjectName(String projectName) {
```

Comment:
should avoid setters whenever possible, as it kind of exposes the internals and makes it easier to unintentionally couple logic from classes and create weird bugs. It's better to create the committer lazily in the OutputFormat, unless there's some reason that doesn't work?

Reply:
Good point, I don't know why I didn't thought of this. Let me make sure that it works and I'll update this logic.

Reply:
Updated. I tested it and it works as expected.